packages/components/app/styles/components/dialog-primitive.scss

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// DIALOG PRIMITIVE
+//
+
+// TEMP STYLES FOR TESTING...
+.hds-dialog-primitive__header,
+.hds-dialog-primitive__description,
+.hds-dialog-primitive__body,
+.hds-dialog-primitive__footer {
+  // outline: 1px dashed red;
+  // background: pink !important;
+}
+
+.hds-dialog-primitive__icon,
+.hds-dialog-primitive__title,
+.hds-dialog-primitive__tagline,
+.hds-dialog-primitive__dismiss {
+  // outline: 1px dashed skyblue !important;
+}
+
+
+// HEADER
+
+.hds-dialog-primitive__header {
+  display: flex;
+  flex: none;
+  gap: 16px;
+  align-items: flex-start;
+  padding: 16px 24px;
+  color: var(--token-color-foreground-strong); // default color
+}
+
+/*
+HEADER DIFFERENCES:
+.hds-modal__header {
+  border-top-left-radius: inherit;
+  border-top-right-radius: inherit;
+}
+
+.hds-flyout__header {}
+*/
```

Comment:
This is just here as a temporary reference while I work on this. It will be deleted in the end.